src/js/components/point-list.js

```diff
@@ -8,20 +8,33 @@ import { display } from 'btc-models/lib/schema/types';
 export class PointList extends Component {
   render() {
     const points = this.props.points.map( point => {
+      let actionButton = <span />;
+      if ( this.props.buttonIcon != undefined ) {
+        actionButton = (
+          <IconButton onClick={ this.props.buttonAction.bind( this, point ) }>
+            <FontIcon className="material-icons"
+              style={ { fontSize: '36px' } }
+              color="red">
+              { this.props.buttonIcon }
+            </FontIcon>
+          </IconButton>
+        );
+      }
+
       return (
-        <Card key={ point._id }
+        <ListItem key={ point._id }
           style={ { margin: '10px' } }
-          onClick={ this.props.onPointClick.bind( this, point ) }>
-          <CardHeader title={ point.name }
-            subtitle={ display( point.type ) }
-            avatar={ point.coverUrl } />
-        </Card>
+          onTouchTap={ this.props.clickAction.bind( this, point ) }
+          leftAvatar={ point.coverUrl }
+          primaryText={ point.name }
+          secondaryText={ display( point.type ) }
+          rightIconButton={ actionButton } />
```

Comment:
We can change this so we don't put an empty <span /> in the list items when there is no button.
```
const actionProps = {};
if( this.props.actionButton ) {
  actionProps.rightIconButton = <Jsx />;
}

<ListItem key={ point._id } { ...actionProps } />
```